app/src/main/java/com/example/android/sunshine/app/LocationEditTextPreference.java

```diff
@@ -52,6 +62,34 @@ private void init(Context context, AttributeSet attrs) {
                 a.recycle();
             }
         }
+
+        GoogleApiAvailability apiAvailability = GoogleApiAvailability.getInstance();
+        int resultCode = apiAvailability.isGooglePlayServicesAvailable(context);
+        if (resultCode == ConnectionResult.SUCCESS) {
+            // Add the get current location widget to our location preference
+            setWidgetLayoutResource(R.layout.pref_current_location);
+        }
```

Comment:
even just a `Log.e` in the else will help you in the future. 

Imagine you forget this needs GooglePlayServices and in a few months you run it on your _amazon phone_ and think _wth_ why isn't it working :-)